### PR TITLE
feat: Init expressjs app and openapi

### PR DESCRIPTION
**Pull request by wmoussa-gc**
Enable basic routing using express-openapi and path versioning.

First Endpoints:

`http://localhost:3000/v1/api-definition`
`http://localhost:3000/v1/forms/{id}/submission/new`

### CODE REVIEW

File: api/v1/routes/forms/{id}/submission/new.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  function GET(req, res, next) {
+    const submission = {
+      id: req.params.id,
+      submission: "My form response",
+    }
+    res.status(200).json(submission);
+  }
+
+  GET.apiDoc = {
+    summary: "Retrieves the oldest submission with status ‘new’ and modifies status to ‘downloaded’",
```
Comment (bryan-robitaille) [inline api/v1/routes/forms/{id}/submission/new.js:15]: Just left a comment in Design Doc 19.  With the issues we're seeing around the indexes in DynamoDB we may have to change the behavior of this endpoint.

**Comment by wmoussa-gc** [comment]
> Did you consider adding Typescript support? I know that is something I did right away when I made a PoC with Express for HTTP streaming. Did not think about not having it.

correct, in the plan

**Comment by craigzour** [comment]
Is the `express-openapi` package just here to generate some kind of an API documentation? Otherwise, when building my PoC (I mentioned in the previous comment), I felt like Express had enough stuff to start building an API.

**Comment by wmoussa-gc** [comment]
> Is the `express-openapi` package just here to generate some kind of an API documentation? Otherwise, when building my PoC (I mentioned in the previous comment), I felt like Express had enough stuff to start building an API.

Not an expert, but express-openapi does more than automatic doc: It provides request and response validation against the OpenAPI schema, routing,  help generate client SDKs, server stubs etc.

**Comment by craigzour** [comment]
> > Is the `express-openapi` package just here to generate some kind of an API documentation? Otherwise, when building my PoC (I mentioned in the previous comment), I felt like Express had enough stuff to start building an API.
> 
> Not an expert, but express-openapi does more than automatic doc: It provides request and response validation against the OpenAPI schema, routing, help generate client SDKs, server stubs etc.

Sounds good! Will read about this when I have time but I guess this is just me trying to make sure we include packages that we know we will need, use and won't hinder our progression towards the API MVP.

**Review by craigzour — APPROVED** [review]

**Pull request merged**